astropy/modeling/core.py

```diff
@@ -727,6 +720,19 @@ def __init__(self, *args, meta=None, name=None, **kwargs):
         self._initialize_slices()
         self._initialize_unit_support()
 
+        # Raise DeprecationWarning on classes with class attributes
+        # ``inputs`` and ``outputs``.
+        self._inputs_deprecation()
+
+    def _inputs_deprecation(self):
```

Comment:
I'm easily confused :-) The following if statement checks for class level n_inputs and then if it exists says the class level inputs is deprecated. Naively I  would have thought the if statement would look for the existence of 'inputs'. Is this check implied elsewhere (e.g, the existence of n_inputs means inputs is defined at the class level, but I don't see that in the recent changes. (and as @Cadair points out defnie should be fixed)

Reply:
Both work, just different logic :-). I changed it and it's clearer.